contracts/executable/AxelarGMPExecutable.sol

```diff
@@ -68,7 +69,7 @@ abstract contract AxelarGMPExecutable is IAxelarGMPExecutable {
      * @notice Returns the address of the AxelarGMPGateway contract.
      * @return The Axelar GMP Gateway instance.
      */
-    function gateway() public view returns (IAxelarGMPGateway) {
-        return IAxelarGMPGateway(gatewayAddress);
+    function gateway() public view returns (IAxelarGMPGatewayWithToken) {
```

Comment:
```suggestion
    function gateway() public view returns (IAxelarGMPGateway) {
```
Again, this executable is only for doing GMP, not GMP calls with tokens. The other executable is used for it

Reply:
updated in next revision.